unit-tests/Test/Booster/SMT/LowLevel.hs

```diff
@@ -0,0 +1,131 @@
+{- |
+Copyright   : (c) Runtime Verification, 2022
+License     : BSD-3-Clause
+-}
+module Test.Booster.SMT.LowLevel (
+    test_smoke,
+) where
+
+import Control.Monad.Logger (runNoLoggingT)
+import Data.ByteString.Char8 qualified as BS
+import Test.Tasty
+import Test.Tasty.HUnit
+
+import Booster.SMT.Base as SMT
+import Booster.SMT.LowLevelCodec
+import Booster.SMT.Runner
+
+test_smoke :: TestTree
+test_smoke =
+    testGroup
+        "Smoke test for SMT bindings (low-level)"
+        [ emptyIsSat
+        , declTests
+        , codecTests
+        , checkTests
+        ]
+
+----------------------------------------
+emptyIsSat :: TestTree
+emptyIsSat =
+    testCase "An empty solver responds to (check-sat) with sat" $
+        (Sat @=?) =<< runSatAfter []
+
+declTests :: TestTree
+declTests =
+    testGroup
+        "Tests for declarations"
+        [ testCase "declare-const" $ runsOK [Declare $ DeclareConst "x" smtInt]
+        , testCase "declare-fun(_)" $ runsOK [Declare $ DeclareFunc "f" [smtInt] smtInt]
+        , testCase "declare-fun(_, _)" $ runsOK [Declare $ DeclareFunc "g" [smtInt, smtInt] smtInt]
+        , testCase "declare simple sort" $ runsOK [Declare $ DeclareSort "SomeSort" 0]
+        , testCase "declare simple sort" $ runsOK [Declare $ DeclareSort "OtherSort" 1]
+        , testCase "declare custom-sorted function" $
+            let sortName = "CustomSort"
+             in runsOK
+                    [ Declare $ DeclareSort sortName 1
+                    , Declare $ DeclareFunc "f" [SMTSort sortName [smtInt]] smtInt
+                    ]
+        , testCase "declare assertion" $ runsOK [Declare $ Assert $ eq 0 0]
```

Comment:
maybe that forall quantification test should go here for testing whether cvc5 is viable?